lib/resty/lock.lua

```diff
@@ -100,8 +102,12 @@ function _M.new(_, dict_name, opts)
         exptime = 30
     end
 
-    if timeout and timeout > exptime then
-        timeout = exptime
+    if timeout then
+        timeout = min(timeout, exptime)
+
+        if step then
+            step = min(step, timeout)
+        end
```

Comment:
> maybe the following can be simpler?

Was my thought too but I didn't want to disturb the part of the code where defaults are set, which are grouped together in the `self` table creation. I'm fine with it though.

> But, we got some logic changed here: timeout will always less than exptime.
> original, timeout may large than exptime

I actually thought this was more something of an overlook, and that the new behavior is beneficial? The doc states:

> This option value cannot be bigger than exptime

Yet, as demonstrated by your example, it can 😱